### PR TITLE
Queue Attorney check out | Delete an issue

### LINKED ISSUE #4478: Attorney check out | Delete an issue
When an attorney checks out a draft decision to a judge, they want to be able to delete any irrelevant issues, but be prevented from deleting any by accident, so that the record reflects the most up to date data about a Veteran's appeal. 

### Acceptance Criteria
- Verify that when a user clicks delete issue from the "Edit Issue" page a modal with the title 'Delete Issue" appears
- Verify that the modal has the following copy: "You are about to permanently delete this issue. To delete please click the "Delete issue" button or click the "Go back link"  to return the previous screen. "
- Verify that the modal has a link text on the left with the following copy "Close" 
- Verify that there is a red secondary button labeled "Delete issue" 
- Verify that the Deleted issues no longer exist in VACOLS when the user confirms the delete within the modal

### Style guide components used:
- Modal

### Mocks and specs
delete issue

### PR DESCRIPTION
**Pull request by dannysellers**
Resolves #4478 

### Description
Adds the Delete Issue flow

### Acceptance Criteria 
- [ ] Verify that when a user clicks delete issue from the "Edit Issue" page a modal with the title 'Delete Issue" appears
- [ ] Verify that the modal has the following copy: "You are about to permanently delete this issue. To delete please click the "Delete issue" button or click the "Go back link"  to return the previous screen. "
- [ ] Verify that the modal has a link text on the left with the following copy "Close" 
- [ ] Verify that there is a red secondary button labeled "Delete issue" 
- [ ] Verify that the Deleted issues no longer exist in VACOLS when the user confirms the delete within the modal

### Testing Plan
1. Go to the Draft Decision checkout flow, Edit Issue.
2. Click Delete issue, verify modal appears.
3. Click Delete issue, verify you're taken back to `/queue/tasks/#{appeal_id}/dispositions`, minus 1 issue, with delete success banner visible.

### CODE REVIEW

**Comment by dannysellers** [comment]
1


2


3

File: client/app/queue/AddEditIssueView.jsx [hunk 1]
```diff
@@ -35,6 +38,12 @@ const smallBottomMargin = css({ marginBottom: '1rem' });
 const noLeftPadding = css({ paddingLeft: 0 });
 
 class AddEditIssueView extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = { modal: false };
```
Comment (mdbenjam) [inline client/app/queue/AddEditIssueView.jsx]: Why use local state here?

File: client/app/queue/AddEditIssueView.jsx [hunk 2]
```diff
@@ -122,6 +131,27 @@ class AddEditIssueView extends React.Component {
...
+      issueId
+    } = this.props;
+    const issueIndex = _.map(issues, 'vacols_sequence_id').indexOf(issue.vacols_sequence_id);
+
+    this.hideDeleteModal();
+
+    this.props.requestDelete(
+      `/appeals/${appeal.id}/issues/${issue.vacols_sequence_id}`, {},
+      `You have deleted issue ${issueIndex + 1}.`
+    ).then(() => this.props.deleteAppealIssue(vacolsId, issueId));
```
Comment (mdbenjam) [inline client/app/queue/AddEditIssueView.jsx]: What do we do on an error?

Comment (dannysellers) [reply client/app/queue/AddEditIssueView.jsx]: On error we dispatch [saveFailure](https://github.com/department-of-veterans-affairs/caseflow/blob/master/client/app/queue/uiReducer/uiActions.js#L73), which sets `state.message[msgType]`. This view is already set up to display errors, through `this.props.error`. In terms of what happens within this page specifically: nothing (besides showing the error). The action is canceled and the page/state doesn't change

File: client/app/queue/AddEditIssueView.jsx (hunk 1 above)
Comment (dannysellers) [reply client/app/queue/AddEditIssueView.jsx]: Ease. I see we use redux for modals elsewhere... updating

File: client/app/queue/AddEditIssueView.jsx (hunk 2 above)
Comment (mdbenjam) [reply client/app/queue/AddEditIssueView.jsx]: Ah awesome, I didn't follow that code path.

**Pull request merged**